### PR TITLE
Added forms forms to create, update and delete books and authors

### PR DESCRIPTION
**Pull request by RizwanNiazFolio3**
### Current Status
Currently the website does not have any way of taking input data from users to change or update the database

### Changes/Additions
A form was added to allow librarian users to renew books
RenewBooks

This form can be used to change the due by date of books borrowed by users, if the entered date is more than 4 weeks away, is in the past, or is not a valid date, then an appropriate error message will be shown.

Invalid3

Another change was the addition of create
Create

update
update

or deletion pages for authors
delete

These pages can only be accessed by Librarians and their links were added to the author_list and author_details pages
authorlistupdate

AuthorDetailsupdate

The same changes were made for the Book class as well
createbook
bookdelete
booklistupdate
Bookdetailsupdate

### Feedback
I'd like general feedback on the code and pull request itself, any changes I should add to the code to make sure it follows the style guide. Best practices etc.

### CODE REVIEW

**Comment by shahabejaz** [comment]
> ### Current Status
> Currently the website does not have any way of taking input data from users to change or update the database
> 
> ### Changes/Additions
> A form was added to allow librarian users to renew books
> RenewBooks
> 
> This form can be used to change the due by date of books borrowed by users, if the entered date is more than 4 weeks away, is in the past, or is not a valid date, then an appropriate error message will be shown.
> 
> Invalid3
> 
> Another change was the addition of create
> Create
> 
> update
> update
> 
> or deletion pages for authors
> delete
> 
> These pages can only be accessed by Librarians and their links were added to the author_list and author_details pages
> authorlistupdate
> 
> AuthorDetailsupdate
> 
> The same changes were made for the Book class as well
> createbook
> bookdelete
> booklistupdate
> Bookdetailsupdate
> 
> ### Feedback
> I'd like general feedback on the code and pull request itself, any changes I should add to the code to make sure it follows the style guide. Best practices etc.

Awesome description!

File: catalog/forms.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import datetime
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
```
Comment (shahabejaz) [inline catalog/forms.py:10]: Type hint and doc string would be useful here.

File: catalog/views.py [hunk 2]
```diff
@@ -9,6 +9,20 @@
 
 from django.contrib.auth.mixins import PermissionRequiredMixin
 
+import datetime
+
```
Comment (shahabejaz) [inline catalog/views.py:13]: we should not add spaces in between imports.

**Review by shahabejaz — COMMENTED** [review]
Improve formatting and type hinting, otherwise looks good.

File: catalog/forms.py (hunk 1 above)
Comment (RizwanNiazFolio3) [reply catalog/forms.py]: Done.

File: catalog/views.py (hunk 2 above)
Comment (RizwanNiazFolio3) [reply catalog/views.py]: Done.

**Review by shahabejaz — APPROVED** [review]
Looks good to me, you just need to resolve conflicts before merging this PR.

**Pull request merged**